internal/scaffold/olm-catalog/csv.go

```diff
@@ -237,6 +237,24 @@ func (s *CSV) setCSVDefaultFields(csv *olmapiv1alpha1.ClusterServiceVersion) {
 			{Type: olmapiv1alpha1.InstallModeTypeAllNamespaces, Supported: true},
 		}
 	}
+	if len(csv.Spec.Icon) == 0 {
+		emptyIcon := olmapiv1alpha1.Icon{
+			Data:      "",
+			MediaType: "",
+		}
+		csv.Spec.Icon = []olmapiv1alpha1.Icon{emptyIcon}
```

Comment:
This will automatically construct an empty Icon:
```suggestion
		csv.Spec.Icon = make([]olmapiv1alpha1.Icon, 1)
```
And remove the above 4 lines.